### PR TITLE
Fix cover fees edit gift

### PR DESCRIPTION
**Pull request by wrandall22**
This PR attempts to fix issues like [this one](https://rollbar.com/Cru/give-web/items/58574/occurrences/147101637267/). I can't reproduce it in a site. I don't know if my test is actually mimicking the issue effectively.

### CODE REVIEW

File: src/common/services/api/cart.service.js [hunk 1]
```diff
@@ -162,8 +163,10 @@ class Cart {
   }
 
   editItem (oldUri, uri, data) {
-    return this.deleteItem(oldUri)
-      .switchMap(() => this.addItem(uri, data, true))
+    return Observable.concat(
+      this.deleteItem(oldUri),
+      this.addItem(uri, data, true)
```
Comment (TheNoodleMoose) [inline src/common/services/api/cart.service.js:168]: The ```concat``` just prevents ```this.addItem``` from starting until ```this.deleteItem``` finishes right?

**Review by TheNoodleMoose — APPROVED** [review]
@wrandall22 I don't seem to have access to the rollbar error, but from what you explained about how ```additem``` was firing before the ```deleteItem``` function finished, this solution makes sense.

**Comment by wrandall22** [comment]
What is your username in rollbar?

**Comment by TheNoodleMoose** [comment]
> What is your username in rollbar?

TheNoodleMoose

**Comment by wrandall22** [comment]
You should have access now.

**Comment by TheNoodleMoose** [comment]
@wrandall22 I still don't seem to have access to it. I have access to https://rollbar.com/Cru/give/ but not https://rollbar.com/Cru/give-web/.

**Comment by wrandall22** [comment]
Ok, should be there now.

File: src/common/services/api/cart.service.js (hunk 1 above)
Comment (wrandall22) [reply src/common/services/api/cart.service.js:168]: That's my understanding. See the [docs](https://www.learnrxjs.io/learn-rxjs/operators/combination/concat).

**Comment by wrandall22** [comment]
Before the fix commit:

```
console.log src/common/services/api/order.service.spec.js:727
    Deleting item

console.log src/common/services/api/order.service.spec.js:727
    Deleting item

console.log src/common/services/api/order.service.spec.js:737
    GET purchase form

  console.log src/common/services/api/order.service.spec.js:731
    Adding item

  console.log src/common/services/api/order.service.spec.js:731
    Adding item

  console.log src/common/services/api/order.service.spec.js:741
    POST purchase
```

After the fix commit:
```
console.log src/common/services/api/order.service.spec.js:727
    Deleting item

  console.log src/common/services/api/order.service.spec.js:731
    Adding item

  console.log src/common/services/api/order.service.spec.js:727
    Deleting item

  console.log src/common/services/api/order.service.spec.js:731
    Adding item

  console.log src/common/services/api/order.service.spec.js:737
    GET purchase form

  console.log src/common/services/api/order.service.spec.js:741
    POST purchase
```

**Comment by TheNoodleMoose** [comment]
Looks good to me!

**Review by TheNoodleMoose — APPROVED** [review]

**Review by TheNoodleMoose — APPROVED** [review]

File: src/app/checkout/checkout.component.js [hunk 2]
```diff
@@ -112,6 +112,7 @@ class CheckoutController {
         })
         .subscribe((data) => {
           this.orderService.addFeesToNewGiftIfNecessary(data)
+          this.orderService.storeCartData(data)
```
Comment (wrandall22) [inline src/app/checkout/checkout.component.js:115]: So, `addFeesToNewGiftIfNecessary` actually calls `storeCartData` so I need to remove these that I added.

Comment (TheNoodleMoose) [reply src/app/checkout/checkout.component.js:115]: Ah yep in the ```updatePrices``` function.

**Review by TheNoodleMoose — APPROVED** [review]

**Comment by wrandall22** [comment]
More problems found: can't edit a gift in my cart.

**Comment by TheNoodleMoose** [comment]
Is that a result of these changes or related to the sync issue with the backend that we are trying to resolve?

**Comment by wrandall22** [comment]
I believe it is a result of my `concat` commit.

**Comment by wrandall22** [comment]
It is deleting the item successfully, but failing on the add.

**Comment by wrandall22** [comment]
A side effect of the `concat` is that briefly the cart shows up as empty when editing a gift.

**Review by TheNoodleMoose — APPROVED** [review]
Is the cart appearing empty too noticeable or do you think it's okay from a user perspective?

**Comment by wrandall22** [comment]
It's not horrible. However, I have another guess at what might be happening here, which would be, if the last item of the cart is removed, maybe the cart itself is deleted and prevents adding another item to it. This would be an unfortunate side effect of doing a delete/add vs a put when changing things in the cart. However, I get 403 responses whenever I try to do a put on an item in the cart, so I can't just switch to that.

**Comment by wrandall22** [comment]
Going to close this PR. I plan to implement the API version of cover fees this week if I can manage it.

**Pull request closed without merging**